### PR TITLE
Only label ready to review pull requests

### PR DESCRIPTION
**Pull request by RachalCassity**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Summary

[Label Pull Request Reviewers workflow ](https://github.com/department-of-veterans-affairs/vets-api/blob/master/.github/workflows/require_backend_label.yml)is supposed to omit backend approval when the requested PR team belongs to Mobile, Identity, or Lighthouse. Unfortunately, labels are not correctly working.

When a pull request is in draft mode:
1. "ready-for-backend-review" label is removed
2. "require-backend-review" label is removed
3. "omit-backend-review" is removed
4. Require backend-review-group approval should fail

## Related issue(s)

- [issue](https://github.com/department-of-veterans-affairs/va.gov-team/issues/92923)

## Testing done

- [ ] *New code is covered by unit tests*
- *Describe what the old behavior was prior to the change*
- *Describe the steps required to verify your changes are working as expected. Exclusively stating 'Specs run' is NOT acceptable as appropriate testing*
- *If this work is behind a flipper:*
  - *Tests need to be written for both the flipper on and flipper off scenarios. [Docs](https://depo-platform-documentation.scrollhelp.site/developer-docs/feature-toggles-guide#Featuretogglesguide-Backendexample).*
  - *What is the testing plan for rolling out the feature?*

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: .github/workflows/code_checks.yml [hunk 1]
```diff
@@ -184,10 +184,36 @@ jobs:
           labels: |
             test-passing
 
+      - name: Add omit-backend-approval label
```
Comment (RachalCassity) [inline .github/workflows/code_checks.yml:187]: Ensure PR is not in draft mode before labeling a PR

File: .github/workflows/code_checks.yml [hunk 2]
```diff
@@ -227,4 +253,4 @@ jobs:
           type: simplecov
           result_path: Coverage Report/.last_run.json
           min_coverage: 90
-          token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (RachalCassity) [inline .github/workflows/code_checks.yml:230]: Nothing was removed

File: .github/workflows/ready_for_review.yml [hunk 3]
```diff
@@ -79,6 +78,7 @@ jobs:
       - name: Add Review label
         uses: actions-ecosystem/action-add-labels@v1
         if: |
+          github.event.pull_request.draft == false &&
```
Comment (RachalCassity) [inline .github/workflows/ready_for_review.yml:81]: Ensure PR is not in draft mode

File: .github/workflows/ready_for_review.yml [hunk 4]
```diff
@@ -90,10 +90,13 @@ jobs:
       - name: Remove Review label
         uses: actions-ecosystem/action-remove-labels@v1
         if: |
-          contains(github.event.pull_request.labels.*.name, 'ready-for-backend-review') &&
-          steps.audit_passing_labels.outputs.ready_for_review != 'true' ||
-          steps.audit_pr_labels.outputs.no_failures != 'true' ||
-          steps.verify_approval.outputs.approval_status == 'confirmed'
+          github.event.pull_request.draft == true ||
```
Comment (RachalCassity) [inline .github/workflows/ready_for_review.yml:93]: Remove review label when pull request is in draft mode

File: .github/workflows/require_backend_label.yml [hunk 5]
```diff
@@ -11,26 +11,40 @@ jobs:
...
+      #   if: github.event.action == 'unlabeled' && github.event.label.name == 'omit-backend-approval'
+      #   uses: actions-ecosystem/action-add-labels@v1
+      #   with:
+      #     number: ${{ github.event.pull_request.number }}
+      #     labels: |
+      #       omit-backend-approval
+
       - name: Add omit-backend-approval label
         if: |
-          !contains(github.event.pull_request.labels.*.name, 'require-backend-approval') &&
```
Comment (RachalCassity) [inline .github/workflows/require_backend_label.yml:32]: Ensure PR is not in draft mode

File: .github/workflows/require_backend_label.yml [hunk 6]
```diff
@@ -45,12 +59,11 @@ jobs:
       - name: Add require-backend-approval label
         uses: actions-ecosystem/action-add-labels@v1
         if: |
-            !contains(github.event.pull_request.labels.*.name, 'require-backend-approval') &&
-            !contains(github.event.pull_request.labels.*.name, 'omit-backend-approval') &&
+            github.event.pull_request.draft == false &&
```
Comment (RachalCassity) [inline .github/workflows/require_backend_label.yml:62]: Ensure PR is not in draft mode

**Review by ryan-mcneil — APPROVED** [review]
🤞

**Review by ryan-mcneil — APPROVED** [review]

**Pull request merged**